Antidote/AbstractCallViewController.m

```diff
@@ -96,25 +98,74 @@ - (void)createSublabel
     [self.topViewContainer addSubview:self.subLabel];
 }
 
+#pragma mark - Autolayout
+
+- (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
+{
+    if (UIInterfaceOrientationIsPortrait(toInterfaceOrientation)) {
+        [self installConstraintsPortrait];
+    }
+    else {
+        [self installConstraintsLandscape];
+    }
+
+    [super willRotateToInterfaceOrientation:toInterfaceOrientation duration:duration];
+}
+
 - (void)installConstraints
 {
-    [self.topViewContainer makeConstraints:^(MASConstraintMaker *make) {
+    if (UIInterfaceOrientationIsPortrait(self.interfaceOrientation)) {
+        [self installConstraintsPortrait];
+    }
+    else {
+        [self installConstraintsLandscape];
+    }
```

Comment:
To avoid code duplication you can move these condition to separate method. Also in this method you can try to avoid more duplication by updating both portrait/landscape constraints.

```
- (void)remakeConstraintsForOrientation:(UIInterfaceOrientation)orientation
{
    BOOL isPortrait = UIInterfaceOrientationIsPortrait(self.interfaceOrientation);

    const CGFloat topViewContainerXOffset = isPortrait ? 0.0 : kIndent;
    const CGFloat topContainerHeight = isPortrait ? kTopContainerHeightPortrait : kTopContainerHeightLandscape;

    [self.topViewContainer remakeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self.view);
        make.left.equalTo(self.view).offset(topViewContainerXOffset);
        make.right.equalTo(self.view).offset(-topViewContainerXOffset);
        make.height.equalTo(topContainerHeight);
    }];

    ...
```